index.bs

```diff
@@ -1118,6 +1118,60 @@ The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "S
 :: A [=credential property=] is some characteristic property of a [=public key credential source=], such as whether it is a
     [=client-side discoverable credential=] or a [=server-side credential=].
 
+: <dfn>Credential Record</dfn>
+:: In order to implement the algorithms defined in [[#sctn-rp-operations]],
+    the [=[RP]=] MUST store some properties of registered [=public key credential sources=].
+    The [=credential record=] [=struct=] is an abstraction of these properties stored in a [=user account=].
+    A&nbsp;credential record is created during a [=registration ceremony=] and used in subsequent [=authentication ceremonies=].
+    [=[RPS]=] MAY delete credential records as necessary or when requested by users.
+
+    The following [=struct/items=] are RECOMMENDED in order to implement all steps of
+    [[#sctn-registering-a-new-credential]] and [[#sctn-verifying-assertion]] as defined:
+
+    <dl dfn-for="credential record">
+        :   <dfn>type</dfn>
+        ::  The [=public key credential source/type=] of the [=public key credential source=].
+
+        :   <dfn>id</dfn>
+        ::  The [=Credential ID=] of the [=public key credential source=].
+
+        :   <dfn>publicKey</dfn>
+        ::  The [=credential public key=] of the [=public key credential source=].
+
+        :   <dfn>signCount</dfn>
+        ::  The latest value of the [=authData/signCount|signature counter=] in the [=authenticator data=]
+            from any [=ceremony=] using the [=public key credential source=].
+
+        :   <dfn>transports</dfn>
+        ::  The value returned from {{AuthenticatorAttestationResponse/getTransports()}}
+            when the [=public key credential source=] was [=registration|registered=].
+
+            Note: Modifying or removing [=list/items=] from the value returned from {{AuthenticatorAttestationResponse/getTransports()}}
+            could negatively impact user experience, or even prevent use of the corresponding credential.
```

Comment:
Reality seems to be the opposite, seems to be that chrome doesn't properly assign all the transports for a credential during registration with getTransports, so removing (clearing) this value is actually safe 😅

Reply:
Chrome had a bug where it was overconfident that it knew a complete set of transports. Chrome 105 will report an empty set unless it has transports from the getInfo or attestation certificate.

Reply:
Setting `transports` to empty in `PublicKeyCredentialDescriptor` is equivalent to omitting it, so yes, that should always work in the sense of not preventing use of a credential. This could make the user experience slightly worse if the user only has `internal`-only credentials, though, as they might get prompted to connect an external authenticator they don't have.

However if you remove items from `transports` but it remains non-empty, then that could prevent using a credential if the client assumes the credential requires an unavailable transport (say, if `getTransports()` returns `["usb", "nfc"]` and you remove `"usb"`, then pass `transports: ["nfc"]` on a device with no NFC antenna).